src/PowerShellEditorServices/Workspace/Workspace.cs

```diff
@@ -301,109 +303,95 @@ public IEnumerable<string> EnumeratePSFiles()
                 try
                 {
-                    subDirs = Directory.GetDirectories(currDir);
+                    psFiles = Directory.GetFiles(folderPath, pattern, SearchOption.TopDirectoryOnly);
                 }
                 catch (DirectoryNotFoundException e)
                 {
                     this.logger.WriteHandledException(
-                        $"Could not enumerate directories in the path '{currDir}' due to it being an invalid path",
+                        $"Could not enumerate files in the path '{folderPath}' due to it being an invalid path",
                         e);
 
                     continue;
                 }
                 catch (PathTooLongException e)
                 {
                     this.logger.WriteHandledException(
-                        $"Could not enumerate directories in the path '{currDir}' due to the path being too long",
+                        $"Could not enumerate files in the path '{folderPath}' due to the path being too long",
                         e);
 
                     continue;
                 }
                 catch (Exception e) when (e is SecurityException || e is UnauthorizedAccessException)
                 {
                     this.logger.WriteHandledException(
-                        $"Could not enumerate directories in the path '{currDir}' due to the path not being accessible",
+                        $"Could not enumerate files in the path '{folderPath}' due to the path not being accessible",
                         e);
 
                     continue;
                 }
 
-                foreach (string subDir in subDirs)
-                {
-                    dirStack.Push(subDir);
-                }
+                foundFiles.AddRange(psFiles);
             }
 
-            return foundFiles;
+            // Prevent unbounded recursion here
+            if (currDepth >= recursionDepthLimit)
+            {
+                this.logger.Write(LogLevel.Warning, $"Recursion depth limit hit for path {folderPath}");
+                return;
+            }
+
+            // Add the recursive directories to search next
+            string[] subDirs;
+            try
+            {
+                subDirs = Directory.GetDirectories(folderPath);
+            }
+            catch (DirectoryNotFoundException e)
+            {
+                this.logger.WriteHandledException(
+                    $"Could not enumerate directories in the path '{folderPath}' due to it being an invalid path",
+                    e);
+
+                return;
+            }
+            catch (PathTooLongException e)
+            {
+                this.logger.WriteHandledException(
+                    $"Could not enumerate directories in the path '{folderPath}' due to the path being too long",
+                    e);
+
+                return;
+            }
+            catch (Exception e) when (e is SecurityException || e is UnauthorizedAccessException)
```

Comment:
I don't have a better suggestion and think this is the right way to do it... but damn these catch statements take up a lot of space.

Reply:
yeah I don't think it's too wide - I meant the 3 catches right after each other twice in this function.

Reply:
I don't like the repetition, but I think a function call around a lambda just to catch certain exceptions is a rather nasty abuse.

Reply:
(Would be more abstractable in a world where exceptions were returned as union types...)